rust/src/lib/nispor/ethernet.rs

```diff
@@ -18,6 +19,29 @@ fn gen_eth_conf(np_iface: &nispor::Iface) -> EthernetConfig {
     if let Some(sriov_info) = &np_iface.sriov {
         eth_conf.sr_iov = Some(gen_sriov_conf(sriov_info));
     }
+    if let Some(sr_iov) = &mut eth_conf.sr_iov {
+        let p: path::PathBuf = [
+            "/sys/class/net",
+            &np_iface.name,
+            "device/sriov_drivers_autoprobe",
+        ]
+        .iter()
+        .collect();
+        if let Ok(contents) = fs::read_to_string(&p) {
+            match contents.as_str().trim().parse::<u8>() {
+                Ok(i) => {
+                    sr_iov.drivers_autoprobe = Some(i == 1);
+                }
+                Err(err) => {
+                    log::warn!(
+                        "failed to parse {}: {:?}",
+                        p.to_string_lossy(),
+                        err
+                    );
+                }
+            }
+        }
+    }
```

Comment:
I am OK to have these code in nmstate project for now. But if possible, please move this code to `nispor` crate at https://github.com/nispor/nispor/

Reply:
I had considered putting the code there but `sriov_drivers_autoprobe` is a PCI level attribute, not a linux network device attribute. It is not exposed by netlink and only via sysfs, it seemed that it didn't have its place in `nispor`.

That makes me think it explains why the integration tests are failing. Since nmstate relies on `nispor` to get the list of VFs associated with a netdev, it will never see VFs created with `drivers_autoprobe = false`. For that, `nmstate` would need to read `sriov_numvfs` in sysfs.

Let me know what you think.

Reply:
nispor currently holds code for querying `sriov_numvfs` even they are in PCI domain. It is doing all heavy lifting on querying network kernel states, nmstate is a consumer here and trust nispor provide accurate data regardless its source.

In nispor, once found `drivers_autoprobe = false`, it could skip searching for VF NIC names. So ideally, nispor should hold the sysfs access code. It could be done later.

Still begging for servers. :(